skopt/dummy_opt.py

```diff
@@ -53,13 +65,19 @@ def dummy_minimize(func, dimensions, maxiter=1000, random_state=None):
     """
     rng = check_random_state(random_state)
     space = Space(dimensions)
-    X = space.rvs(n_samples=maxiter, random_state=rng)
-
-    init_y = func(X[0])
+    if x0 is None:
+        x0 = []
+    if y0 is None:
+        y0 = [func(x) for x in x0]
+    if len(x0) != len(y0):
+        raise ValueError('x0 and y0 should have the same length')
+    X = x0 + space.rvs(n_samples=maxiter, random_state=rng)
+    init_y = func(X[0]) if len(y0) == 0 else y0[0]
```

Comment:
if `len(y0)==0`, does this call `func` once more than permitted? (it is called twice for `func(X[0])` in that case)